app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt

```diff
@@ -124,19 +120,25 @@ class TabsTrayFragment : AppCompatDialogFragment(), TabsTrayInteractor {
             )
 
         tabsTrayController = DefaultTabsTrayController(
-            store = tabsTrayStore,
+            trayStore = tabsTrayStore,
+            browserStore = requireComponents.core.store,
             browsingModeManager = activity.browsingModeManager,
             navController = findNavController(),
+            navigateToHomeAndDeleteSession = ::navigateToHomeAndDeleteSession,
```

Comment:
I went with this separate method because the essence of `navigateToHomeAndDeleteSession` is to save the `sessionId` of the tab to delete in `homeViewModel` and then navigate. Thought this is too important to leave it up to the controller and it's future refactorings so I've encapsulated this two actions in a single method.
We could send the `homeViewModel` to the Controller and rely on tests to ensure the right calls in the right order are made ?

Reply:
Ah, that's a fair point. Let's leave it where it is in that case. 🙂